src/DADynamoApp/DAEntrypoint.cs

```diff
@@ -0,0 +1,182 @@
+            if (proc.HasExited)
+            {
+                Console.WriteLine($"Dynamo exiting with code {proc.ExitCode}");
+            }
+        }
+
+        private static void CurrentDomain_UnhandledException(object sender, UnhandledExceptionEventArgs e)
+        {
+            Console.WriteLine($"Unhandled exception: {e}");
+        }
+
+        public ExternalDBApplicationResult OnStartup(ControlledApplication application)
+        {
+            AppDomain.CurrentDomain.UnhandledException += CurrentDomain_UnhandledException;
+            AppDomain.CurrentDomain.ProcessExit += CurrentDomain_ProcessExit;
+            AppDomain.CurrentDomain.AssemblyResolve += DynamoPathResolver.ResolveAssembly;
+
+            Console.WriteLine("<<!>> Starting to load D4DA");
+
+            try
+            {
+                RevitServices.Transactions.TransactionManager.SetupManager(new RevitServices.Transactions.AutomaticTransactionStrategy());
+                ElementBinder.IsEnabled = true;
+
+                RevitServicesUpdater.Initialize(Updaters);
+
+                DesignAutomationBridge.DesignAutomationReadyEvent += HandleDesignAutomationReadyEvent;
+
+                Console.WriteLine("<<!>> D4DA Loaded");
+
+                return ExternalDBApplicationResult.Succeeded;
+            }
+            catch (Exception ex)
+            {
+                return ExternalDBApplicationResult.Failed;
+            }
+        }
+
+        public void HandleDesignAutomationReadyEvent(object sender, DesignAutomationReadyEventArgs e)
+        {
+            Console.WriteLine("<<!>> DA event raised.");
+            WorkItemFolder = Directory.GetCurrentDirectory();
+
+            foreach (var xx in Directory.EnumerateFiles(WorkItemFolder))
+                Console.WriteLine(xx);
+
+            var dynTempDir = Path.Combine(WorkItemFolder, "dyn_tmp");
+            Console.WriteLine($"<<!>> Work folder is '{WorkItemFolder}'");
+            var app = e.DesignAutomationData?.RevitApp;
+            Console.WriteLine("<<!>> Preparing Dynamo model. Vers 1");
+
+            var hostloc = typeof(Autodesk.Revit.ApplicationServices.Application).Assembly.Location;
+            var asmLocation = Path.GetDirectoryName(hostloc);
+            Console.WriteLine($"using asm at location {asmLocation}");
+            Console.WriteLine($"Is Loaded {LoadMessage}");
+            var curFolder = Directory.GetParent(Assembly.GetExecutingAssembly().Location).FullName;
+            var cmdArgs = new Dynamo.Applications.StartupUtils.CommandLineArguments();
+            cmdArgs.ASMPath = asmLocation;
+            cmdArgs.DisableAnalytics = true;
+            cmdArgs.ServiceMode = true;
+            cmdArgs.AnalyticsInfo = new() { HostName = "Revit_DA" };
+            cmdArgs.ImportedPaths = [Path.Combine(curFolder, "nodes", "RevitNodes.dll"), Path.Combine(curFolder, "nodes", "DSRevitNodesUI.dll")];
```

Comment:
okay, you're going to need to give me some more info as to why you think this approach is better than what Dynamo/DynamoRevit does normally to resolve dependencies?

Or is the crux, that you want to avoid subclassing DynamoModel for some reason?